lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -520,7 +603,7 @@ void Inliner::cloneAndRename(
 
   // Clone and rename.
   auto *newOp = b.clone(op, mapper);
```

Comment:
In the normal case, can we just splice the old body into the module and fix it up rather than cloning?  Maybe file an issue to fix this later.

Reply:
That seems like a reasonable optimization. Without dedup, this should be the common case. After aggressive dedup, then cloning is likely always necessary. This would require some information that the module being inlined/flattened can be moved, i.e., the module is dead after inlining/flattening. Currently dead module determination happens after the pass runs. I'll open the issue.